sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -289,26 +303,7 @@ def transform(self, X, y=None):
         Xa : {array, sparse matrix}
             Feature vectors; always 2-d.
         """
-        if self.sparse:
-            return self._transform(X, fitting=False)
-
-        else:
-            dtype = self.dtype
-            vocab = self.vocabulary_
-            X = _tosequence(X)
-            Xa = np.zeros((len(X), len(vocab)), dtype=dtype)
-
-            for i, x in enumerate(X):
-                for f, v in six.iteritems(x):
-                    if isinstance(v, six.string_types):
-                        f = "%s%s%s" % (f, self.separator, v)
-                        v = 1
-                    try:
-                        Xa[i, vocab[f]] = dtype(v)
-                    except KeyError:
-                        pass
-
-            return Xa
+        return self._transform(X, fitting=False)
```

Comment:
What's happened to the dense case?

Reply:
it's taken care of here https://github.com/scikit-learn/scikit-learn/pull/8750/files/488562d95c92dd03a2b283eb0580e15a199ee2aa#diff-db0cc7cd94253605c2325da001c2983aR219